### PR TITLE
Add draft for august

### PR DESCRIPTION
**Pull request by Hoverbear**

### CODE REVIEW

**Comment by Hoverbear** [comment]
PTAL @nrc @dcalvin

File: august-19.md [hunk 1]
```diff
@@ -1,62 +1,151 @@
...
-TODO:
+We've also began nailing down which features we want to include in a future TiKV 4.0 release. Here's a few of our hopes and dreams:
 
-* backup/restore
-* joint consensus
-* pessimistic transactions
+> (This is not an official list and may be subject to change)
 
-## RFCs
+* Publish our RocksDB wrapper as `tirocks` on [crates.io](https://crates.io/crates/tirocks/).
```
Comment (siddontang) [inline august-19.md:44]: seem we have not published it yet? only occupy the name

File: august-19.md [hunk 2]
```diff
@@ -1,62 +1,151 @@
...
-
-* [2019-07-08](https://pingcap.com/weekly/2019-07-08-tidb-weekly/)
-* ...
-
+* [2019-08-04](https://pingcap.com/weekly/2019-08-05-tidb-weekly/)
+* [2019-08-12](https://pingcap.com/weekly/2019-08-12-tidb-weekly/)
+* [2019-08-19](https://pingcap.com/weekly/2019-08-19-tidb-weekly/)
+* [2019-08-26](https://pingcap.com/weekly/2019-08-26-tidb-weekly/)
+
+[@sunxiaoguang]: https://github.com/
```
Comment (MyonKeminta) [inline august-19.md]: Seems the username part is missing

File: august-19.md [hunk 3]
```diff
@@ -1,62 +1,151 @@
...
+* [@sticnarf] and [@nrc] have been working on transaction support in our Rust Client.
+* [@fullstop000] has been working to implement follower replication in Raft, for future use in TiKV.
+* [@hicqu]has been leading the effort on supporting Joint Consensus in TiKV.
+* Several members of the [@pingcap] team have been working on further shrinking and optimizing TiKV's build graph, resulting in faster compile times and less dependencies.
+* [@zhouqiang-cl] and [@hoverbear] have been working on improving our release process and docker containers.
+* [@dcalvin] and [@hoverbear] have been overhauling the TiKV.org documentation to better support our users.
+* [@busyjay] and [@nrc] have been collaborating to land the final PRs to support using [Prost](https://github.com/danburkert/prost) library.
+* [@nrc] has been working on Prost to help land some optimizations.
+* [@overvenus] has been driving forward our new backup and restore process for TiKV.
+* [@MyonKeminta] has been leading on refining and hardening our experimental pessimistic transaction support.
```
Comment (MyonKeminta) [inline august-19.md:87]: I think it's @youjiali1995 who leads pessimistic transaction.

File: august-19.md (hunk 2 above)
Comment (dcalvin) [inline august-19.md]:
```suggestion
[@sunxiaoguang]: https://github.com/sunxiaoguang/
```

File: august-19.md [hunk 4]
```diff
@@ -1,62 +1,151 @@
 # This month in TiKV: August 2019
 
-TODO
+Happy day! Welcome to the second edition of 'This Month in TiKV', covering August 2019. (Yes we know it's later than the first week of September, sorry!)
 
-News and free-form section
-
-* chat/welcome
-* success stories - adoption, new clients, etc.
+The TiKV authors have been busy working on improving stability, fixing bugs, and laying out the foundations of TiKV 4.0. That's a lot to cover so lets get started!
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
The TiKV authors have been busy working on improving stability, fixing bugs, and laying out the foundations of TiKV 4.0. That is a lot to cover, so let's get started!
```

File: august-19.md [hunk 5]
```diff
@@ -1,62 +1,151 @@
...
 
 ## News
 
-TODO
+This month our team made four TiKV releases! These minor releases include bugfixes and minor, backwards compatible features.
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
This month our team made four small TiKV releases! These releases include bug fixes and minor, backwards compatible features.
```

File: august-19.md [hunk 6]
```diff
@@ -1,62 +1,151 @@
...
+* [2.1.16](https://github.com/tikv/tikv/releases/tag/v2.1.16)
 
+Upgrading? Please take note of these things:
 
-## Events
+* In **2.1.17 and 3.0.3** a new `config-check` option allows you to verify a configuration is correct before trying to start a node with it.
+* In **3.0.2** some logs which require no manual intervention (TiKV resolves them automatically) were reduced to `INFO` level.
+* In **2.1.16** `raw_scan` and `raw_batch_scan` now supports reverse.
 
 ## Reading material
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
## Reading materials
```

File: august-19.md [hunk 7]
```diff
@@ -1,62 +1,151 @@
...
 
-TODO: Dongxu posts - pessimistic locking and ...
+Here's some things our contributors have published over the last month:
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
Here are some articles our contributors have published over the last month:
```

File: august-19.md [hunk 8]
```diff
@@ -1,62 +1,151 @@
...
 
-## Current projects
+* [@sunxiaoguang] published some details of the deployment including TiKV at Zhihu in [Lesson Learned from Queries over 1.3 Trillion Rows of Data Within Milliseconds of Response Time at Zhihu.com](https://pingcap.com/success-stories/lesson-learned-from-queries-over-1.3-trillion-rows-of-data-within-milliseconds-of-response-time-at-zhihu/), some highlights from article:
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
* [@sunxiaoguang] published some details of the deployment including TiKV at Zhihu in [Lesson Learned from Queries over 1.3 Trillion Rows of Data Within Milliseconds of Response Time at Zhihu.com](https://pingcap.com/success-stories/lesson-learned-from-queries-over-1.3-trillion-rows-of-data-within-milliseconds-of-response-time-at-zhihu/). Some highlights from the article are:
```

File: august-19.md [hunk 9]
```diff
@@ -1,62 +1,151 @@
...
-* backup/restore
-* joint consensus
-* pessimistic transactions
+> (This is not an official list and may be subject to change)
 
-## RFCs
+* Publish our RocksDB wrapper as `tirocks` on [crates.io](https://crates.io/crates/tirocks/).
+* Abstract our engine so that we adapt to other storage technologies and offer more flexibility.
+* Begin adopting Joint Consensus, first to allow our clusters to more safely replace nodes in running deployments.
+* Further battle-hardening and improvements to the titan storage backend.
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
* Further battle-hardening and improvements to the Titan storage backend.
```

File: august-19.md [hunk 10]
```diff
@@ -1,62 +1,151 @@
...
+* Reduce WAL usage to lower write pressure.
 
 ## Notable PRs
 
+* In TiKV 2.1.17 gRPC was upgraded to fix a segfault causing abnormal exits [#5441](https://github.com/tikv/tikv/pull/5441).
+* In TiKV 2.1.17 and 3.0.3 incorrect timestamps should no longer be reported from a Region [#5296](https://github.com/tikv/tikv/pull/5296).
+* In TiKV 3.0.3 a possible request drop from ReadIndex when there is no leader was fixed [#5316](https://github.com/tikv/tikv/pull/5316).
+* TiKV 2.1.16 now returns region errors when closing so the client will retry on other nodes [#4820](https://github.com/tikv/tikv/pull/4820).
+* In TiKV 3.0.2 we fixed a bug in constraint checking during insertion when pessimistic transactions are enabled [#5128](https://.github.com/tikv/tikv/pull/5128).
+* In TiKV 3.0.2 we fixed the problem that TiKV lose some logs while panicking [#5174](https://github.com/tikv/tikv/pull/5174).
```
Comment (dcalvin) [inline august-19.md]:
```suggestion
* In TiKV 3.0.2 we fixed the problem that TiKV loses some logs while panicking [#5174](https://github.com/tikv/tikv/pull/5174).
```

File: august-19.md [hunk 11]
```diff
@@ -1,62 +1,151 @@
...
+We've also began nailing down which features we want to include in a future TiKV 4.0 release. Here's a few of our hopes and dreams:
 
-* backup/restore
-* joint consensus
-* pessimistic transactions
+> (This is not an official list and may be subject to change)
 
-## RFCs
+* Publish our RocksDB wrapper as `tirocks` on [crates.io](https://crates.io/crates/tirocks/).
+* Abstract our engine so that we adapt to other storage technologies and offer more flexibility.
```
Comment (siddontang) [inline august-19.md:45]: May using `start to abstrace` is better?

**Comment by dcalvin** [comment]
@Hoverbear Let's resolve the comments and get this out, shall we?

File: august-19.md (hunk 11 above)
Comment (dcalvin) [reply august-19.md:45]: These are Roadmap items - it indicates something just started or to be started.  So I  think it is fine here.

File: august-19.md (hunk 1 above)
Comment (Hoverbear) [reply august-19.md:44]: Yes! That's why it's on the ROADMAP. :)

File: august-19.md (hunk 3 above)
Comment (Hoverbear) [reply august-19.md:87]: Oh! Thank you!

**Comment by Hoverbear** [comment]
@dcalvin Sorry for the delay. I think this is good so I'm going to open a PR on the website with this content now. :)

**Review by nrc — APPROVED** [review]
lgtm, nice work!

File: august-19.md [hunk 12]
```diff
@@ -1,62 +1,154 @@
...
+* [@busyjay] and [@nrc] have been collaborating to land the final PRs to support using [Prost](https://github.com/danburkert/prost) library.
+* [@nrc] has been working on Prost to help land some optimizations.
+* [@overvenus] has been driving forward our new backup and restore process for TiKV.
+* [@MyonKeminta] has been leading on refining and hardening our experimental pessimistic transaction support.
+* [@zhouqiang-cl]'s team at [@pingcap] has been working to improve our CI and automation to help reduce time-to-merge, expand testing, and improve automation.
+* [@yiwu-arbug] and [Connor1996] are battle hardening Titan and incorporating feedback from some of our users.
+* [@sunxiaoguang] has been battle testing TiKV on massive, China-scale workloads at Zhihu ([See his article!](https://pingcap.com/success-stories/lesson-learned-from-queries-over-1.3-trillion-rows-of-data-within-milliseconds-of-response-time-at-zhihu/))
+* [@busyjay] and [@zhangjinpeng1987] have been evaluating bloom filters.
+* [@5kbpers], [@brson] and [@aknuds1] have begun work on the engine abstraction project.
+* [MyonKeminta] has been driving a project to facilitate larger transactions.
```
Comment (brson) [inline august-19.md:93]:
```suggestion
* [@MyonKeminta] has been driving a project to facilitate larger transactions.
```

**Review by brson — APPROVED** [review]

**Comment by brson** [comment]
I did mark one suggested change.

**Comment by brson** [comment]
This is packed with good info.

**Comment by Hoverbear** [comment]
Thanks @brson !

**Comment by Hoverbear** [comment]
@nrc Do you think merging this is necessary? Should we keep using this repo?

**Comment by Hoverbear** [comment]
Ping @nrc

**Comment by nrc** [comment]
> @nrc Do you think merging this is necessary? Should we keep using this repo?

I don't really, but I did anyway. I had hoped people would post to the issue as we go along, but it seems just as easy to write each issue in one go at the end of the month, in which case this repo isn't really necessary.

**Pull request merged**